client/lib/promote-post/index.ts

```diff
@@ -138,17 +139,18 @@ export enum PromoteWidgetStatus {
  * @returns bool
  */
 export const usePromoteWidget = (): PromoteWidgetStatus => {
-	const value = useSelector( ( state ) => {
-		const userData = getCurrentUser( state );
-		if ( userData ) {
-			const originalSetting = userData[ 'has_promote_widget' ];
-			if ( originalSetting !== undefined ) {
-				return originalSetting === true
-					? PromoteWidgetStatus.ENABLED
-					: PromoteWidgetStatus.DISABLED;
-			}
-		}
-		return PromoteWidgetStatus.FETCHING;
-	} );
-	return value;
+	const selectedSite = useSelector( getSelectedSite );
+
+	const value = useSelector( ( state ) =>
+		getSiteOption( state, selectedSite?.ID, 'has_promote_widget' )
+	);
+
+	switch ( value ) {
+		case false:
+			return PromoteWidgetStatus.DISABLED;
+		case true:
+			return PromoteWidgetStatus.ENABLED;
+		default:
```

Comment:
Yeah, agree, that could improve the readability. Maybe it's because it is the end of my day, but I'm not having great ideas to overcome this error without adding another `return PromoteWidgetStatus.FETCHING;` at the end of the function (or adding more code):

![image](https://user-images.githubusercontent.com/1044309/227382804-67083bd7-eda9-48dc-ae52-2b6caa48b0dc.png)

Suggestions?